src/utils/vendor/package-manager.ts

```diff
@@ -11,26 +11,28 @@
  */
 import { execSync } from 'node:child_process'
 import { existsSync } from 'node:fs'
-import { join } from 'node:path'
+import { join, sep } from 'node:path'
 
-export const packageManagers = ['pnpm', 'npm', 'yarn'] as const
+export const packageManagers = ['pnpm', 'npm', 'yarn', 'bun'] as const
 export type PackageManager = (typeof packageManagers)[number]
 
 export function detectInvokedPackageManager(): PackageManager {
-  let detectedPackageManager: PackageManager = 'npm'
-  const invoker = require.main
-
-  if (!invoker) {
-    return detectedPackageManager
+  if (process.env.npm_config_user_agent) {
+    for (const pm of packageManagers) {
+      if (process.env.npm_config_user_agent.startsWith(`${pm}/`)) {
+        return pm
+      }
+    }
   }
 
-  for (const pkgManager of packageManagers) {
-    if (invoker.path.includes(pkgManager)) {
-      detectedPackageManager = pkgManager
-      break
+  if (process.env.npm_execpath) {
+    for (const pm of packageManagers) {
+      if (process.env.npm_execpath.split(sep).includes(pm)) {
+        return pm
+      }
     }
   }
-  return detectedPackageManager
+  return 'npm'
 }
```

Comment:
@beeman i am not sure if this function needs to change or not but this is what nx is using now https://github.com/nrwl/nx/blob/master/packages/create-nx-workspace/src/utils/package-manager.ts

is there a way to test `bun create` locally? i get the error `warn: bun create [local file] only supports .jsx and .tsx files currently`

currently this code is working for `bunx` but otherwise untested

Reply:
I'll take a look at the differences between those.

This is what I'm getting with `bun create`
<img width="691" alt="image" src="https://github.com/user-attachments/assets/456175ae-2793-4f86-a035-bd08ffbeec14" />

Reply:
I just did some local testing and managed to get a working project with `bun`. 

I reckon we can merge it and see how it works in the wild and fix any issues

Reply:
That sounds good to me